pyprof/nvtx/config.py

```diff
@@ -15,18 +15,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-class Config:
-    __instance = None
 
-    @staticmethod
-    def getInstance():
-        if Config.__instance == None:
-            Config()
-        return Config.__instance
-    
-    def __init__(self, **kwargs):
-        if Config.__instance != None:
-            raise Exception("This is a singleton")
-        else:
-            Config.__instance = self
-            self.func_stack_enabled = kwargs.get("enable_function_stack", False)
+class Config(object):
+    _instance = None
+
+    # Overloading the __new__ method enables singleton behavior
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super(Config, cls).__new__(cls)
+            cls.func_stack_enabled = kwargs.get("enable_function_stack",
+                                                False) or kwargs.get("capture_input_ops", False)
+            cls.capture_input_ops = kwargs.get("capture_input_ops", False)
```

Comment:
I still think we should guard against capture_input_ops=true && func_stack_enabled=false here. Either add an assert, or set func_stack_enabled=true if capture_input_ops.

Reply:
(The description of the capture_input_ops feature could be that it explicitly enables func_stack)

Reply:
What do you mean by description?

Reply:
Done